ext/console/02_console.js

```diff
@@ -6,324 +6,611 @@ const core = globalThis.Deno.core;
+  bgGray: [100, defaultBG], // Alias: bgGrey, bgBlackBright
+  bgRedBright: [101, defaultBG],
+  bgGreenBright: [102, defaultBG],
+  bgYellowBright: [103, defaultBG],
+  bgBlueBright: [104, defaultBG],
+  bgMagentaBright: [105, defaultBG],
+  bgCyanBright: [106, defaultBG],
+  bgWhiteBright: [107, defaultBG],
+};
 
-function isInvalidDate(x) {
-  return isNaN(DatePrototypeGetTime(x));
+function defineColorAlias(target, alias) {
+  ObjectDefineProperty(colors, alias, {
+    get() {
+      return this[target];
+    },
+    set(value) {
+      this[target] = value;
+    },
+    configurable: true,
+    enumerable: false,
+  });
 }
 
-function hasOwnProperty(obj, v) {
-  if (obj == null) {
+defineColorAlias("gray", "grey");
+defineColorAlias("gray", "blackBright");
+defineColorAlias("bgGray", "bgGrey");
+defineColorAlias("bgGray", "bgBlackBright");
+defineColorAlias("dim", "faint");
+defineColorAlias("strikethrough", "crossedout");
+defineColorAlias("strikethrough", "strikeThrough");
+defineColorAlias("strikethrough", "crossedOut");
+defineColorAlias("hidden", "conceal");
+defineColorAlias("inverse", "swapColors");
+defineColorAlias("inverse", "swapcolors");
+defineColorAlias("doubleunderline", "doubleUnderline");
+
+// https://tc39.es/ecma262/#sec-boolean.prototype.valueof
+const _booleanValueOf = Boolean.prototype.valueOf;
+
+// https://tc39.es/ecma262/#sec-number.prototype.valueof
+const _numberValueOf = Number.prototype.valueOf;
+
+// https://tc39.es/ecma262/#sec-string.prototype.valueof
+const _stringValueOf = String.prototype.valueOf;
+
+// https://tc39.es/ecma262/#sec-symbol.prototype.valueof
+const _symbolValueOf = Symbol.prototype.valueOf;
+
+// https://tc39.es/ecma262/#sec-weakmap.prototype.has
+const _weakMapHas = WeakMap.prototype.has;
+
+// https://tc39.es/ecma262/#sec-weakset.prototype.has
+const _weakSetHas = WeakSet.prototype.has;
+
+// https://tc39.es/ecma262/#sec-get-arraybuffer.prototype.bytelength
+const _getArrayBufferByteLength = ObjectGetOwnPropertyDescriptor(
+  ArrayBufferPrototype,
+  "byteLength",
+).get;
+
+// https://tc39.es/ecma262/#sec-get-sharedarraybuffer.prototype.bytelength
+let _getSharedArrayBufferByteLength;
+
+// https://tc39.es/ecma262/#sec-get-set.prototype.size
+const _getSetSize = ObjectGetOwnPropertyDescriptor(
+  SetPrototype,
+  "size",
+).get;
+
+// https://tc39.es/ecma262/#sec-get-map.prototype.size
+const _getMapSize = ObjectGetOwnPropertyDescriptor(
+  MapPrototype,
+  "size",
+).get;
+
+function isObjectLike(value) {
+  return value !== null && typeof value === "object";
+}
+
+export function isAnyArrayBuffer(value) {
+  return isArrayBuffer(value) || isSharedArrayBuffer(value);
+}
+
+export function isArgumentsObject(value) {
+  return (
+    isObjectLike(value) &&
+    value[SymbolToStringTag] === undefined &&
+    ObjectPrototypeToString(value) === "[object Arguments]"
+  );
+}
+
+export function isArrayBuffer(value) {
+  try {
+    _getArrayBufferByteLength.call(value);
```

Comment:
I was reading through this PR to see if the `console` code is beginner-friendly enough that I could fix https://github.com/denoland/deno/issues/18828 by myself, and I have a general question :)

I noticed that you get many built-ins from `primordials`, I assume so that user code doesn't mess up with them by modifying the prototypes of the various built-in JS objects. However, isn't `.call` vulnerable? e.g.
```js
// user code
Object.getOwnPropertyDescriptor(ArrayBuffer.prototype, "byteLength").get.call = () => 3;
```
and then `isArrayBuffer` will always return `true`.